packages/@jsii/spec/src/assembly-utils.ts

```diff
@@ -64,13 +64,46 @@ export function writeAssembly(
   return compress;
 }
 
+/**
+ * Loads the assembly file and, if instructed to, redirects to the
+ * compressed assembly file.
+ *
+ * Note: this function does not actually look for the filename specified in
+ * the redirect schema, since we are not dealing with files. Instead, it assumes
+ * that the compressedAssemblyBuffer holds the assembly we are looking for.
+ *
+ * @param assemblyBuffer buffer containing SPEC_FILE_NAME contents
+ * @param compressedAssemblyBuffer buffer containing SPEC_FILE_NAME_COMPRESSED contents, if available
+ * @param validate whether or not to validate the assembly
+ */
+export function loadAssemblyFromBuffer(
+  assemblyBuffer: Buffer,
+  compressedAssemblyBuffer?: Buffer,
```

Comment:
Maybe this could be made to depend on the compressed file name contract by accepting a callback for more files (`(fileName: string) => Buffer`)?

Would likely make it easier/possible to use that implementation in the other `loadAssembly` variants...